src/tiffimage_int.hpp

```diff
@@ -146,29 +146,13 @@ struct TiffImgTagStruct {
   IfdId group_;   //!< Group that contains the image tag
 };                // struct TiffImgTagStruct
 
+typedef std::pair<uint32_t, IfdId> TiffGroupKey;
+
 /*!
   @brief Data structure used as a row (element) of a table (array)
          defining the TIFF component used for each tag in a group.
  */
-struct TiffGroupStruct {
-  //! Search key for TIFF group structure.
-  using Key = std::pair<uint32_t, IfdId>;
-
-  //! Comparison operator to compare a TiffGroupStruct with a TiffGroupStruct::Key
-  bool operator==(const Key& key) const {
-    auto [e, g] = key;
-    return g == group_ && (Tag::all == extendedTag_ || e == extendedTag_);
-  }
-  //! Return the tag corresponding to the extended tag
-  [[nodiscard]] uint16_t tag() const {
-    return static_cast<uint16_t>(extendedTag_ & 0xffff);
-  }
-
-  // DATA
-  uint32_t extendedTag_;           //!< Tag (32 bit so that it can contain special tags)
-  IfdId group_;                    //!< Group that contains the tag
-  NewTiffCompFct newTiffCompFct_;  //!< Function to create the correct TIFF component
-};
+typedef std::map<TiffGroupKey, NewTiffCompFct> TiffGroupTable;
```

Comment:
same here.

Reply:
[question] Did you try to use `unordered_map` instead? In my experience, `unordered_map` is most of the times faster than `map`. Unless you are strictly interested in maintaining the structure sorted, the hash map is normally faster.

Reply:
That's a good idea. I'm running into trouble though because `std::pair<uint32_t, IfdId>` doesn't have a `std::hash` instance. I'm still figuring out how to solve that.

Reply:
Just out of curiosity, did you measure the time after the change to `unordered_map`? I would like to know if the guess was correct or not.